core/include/scipp/core/value_and_variance.h

```diff
@@ -73,6 +73,11 @@ template <class T> constexpr auto abs(const ValueAndVariance<T> a) noexcept {
   return ValueAndVariance{abs(a.value), a.variance};
 }
 
+template <class T> constexpr auto isnan(const ValueAndVariance<T> a) noexcept {
+  using std::isnan;
+  return isnan(a.value);
```

Comment:
Would it make sense to also check the variance and return true if either is NAN?